pkg/lib/bundle/validate.go

```diff
@@ -132,11 +170,159 @@ func (i imageValidator) ValidateBundle(directory string) error {
+		data, err := ioutil.ReadFile(fileWithPath)
+		if err != nil {
+			validationErrors = append(validationErrors, fmt.Errorf("Unable to read file %s in supported types", fileWithPath))
+			continue
+		}
+
+		dec := k8syaml.NewYAMLOrJSONDecoder(strings.NewReader(string(data)), 30)
+		k8sFile := &unstructured.Unstructured{}
+		if err := dec.Decode(k8sFile); err == nil {
+			unstObjs = append(unstObjs, k8sFile)
+			kind := k8sFile.GetObjectKind().GroupVersionKind().Kind
+			i.logger.Debugf(`Validating file "%s" with type "%s"`, item.Name(), kind)
+			// Verify if the object kind is supported for registryV1 format
+			if _, ok := supportedTypes[kind]; !ok {
+				validationErrors = append(validationErrors, fmt.Errorf("%s is not supported type for registryV1 bundle: %s", kind, fileWithPath))
+			} else {
+				if kind == csvKind {
+					csv := &v1.ClusterServiceVersion{}
+					err := runtime.DefaultUnstructuredConverter.FromUnstructured(k8sFile.Object, csv)
+					if err != nil {
+						validationErrors = append(validationErrors, err)
+					}
+
+					csvName = csv.GetName()
+					results := csvValidator.Validate(csv)
+					if len(results) > 0 {
+						for _, err := range results[0].Errors {
+							validationErrors = append(validationErrors, err)
+						}
+					}
+				} else if kind == crdKind {
+					crd := &v1beta1.CustomResourceDefinition{}
+					err := runtime.DefaultUnstructuredConverter.FromUnstructured(k8sFile.Object, crd)
+					if err != nil {
+						validationErrors = append(validationErrors, err)
+					}
+
+					results := crdValidator.Validate(crd)
+					if len(results) > 0 {
+						for _, err := range results[0].Errors {
+							validationErrors = append(validationErrors, err)
+						}
+					}
+				} else {
+					err := validateKubectlable(data)
+					if err != nil {
+						validationErrors = append(validationErrors, err)
+					}
+				}
+			}
+		} else {
+			validationErrors = append(validationErrors, err)
+		}
+	}
+
+	// Validate the bundle object
+	if len(unstObjs) > 0 {
+		bundle := registry.NewBundle(csvName, "", "", unstObjs...)
+		bundleValidator := v.BundleValidator
+		results := bundleValidator.Validate(bundle)
+		if len(results) > 0 {
+			for _, err := range results[0].Errors {
+				validationErrors = append(validationErrors, err)
+			}
+		}
+	}
+
+	if len(validationErrors) > 0 {
+		return NewValidationError(validationErrors)
+	}
+
+	return nil
+}
+
+// Validate if the file is kubecle-able
+func validateKubectlable(fileBytes []byte) error {
+	exampleFileBytesJson, err := y.YAMLToJSON(fileBytes)
+	if err != nil {
+		return err
+	}
+
+	parsedMeta := &Meta{}
+	err = json.Unmarshal(exampleFileBytesJson, parsedMeta)
+	if err != nil {
+		return err
+	}
+
+	errs := apiValidation.ValidateObjectMeta(
+		&parsedMeta.ObjectMeta,
+		false,
+		func(s string, prefix bool) []string {
+			return nil
+		},
+		field.NewPath("metadata"),
+	)
+
+	if len(errs) > 0 {
+		return fmt.Errorf("error validating object metadata: %s. %v", errs, parsedMeta)
```

Comment:
should unpack the errs type `fields.ErrorList` with something like `errs.ToAggregate()` so it looks cleaner to the end user?